idaes/core/util/initialization.py

```diff
@@ -189,7 +191,7 @@ def propagate_state(destination=None, source=None, arc=None,
     for k, v in destination_vars.items():
         try:
             for i in v:
-                if not (v[i].is_variable_type() and v[i].fixed):
+                if v[i].is_variable_type() and ((not v[i].fixed) or overwrite_fixed):
```

Comment:
The behavior got changed for when `v[i].is_variable_type()` is false. Are there things for which `v[i].is_variable_type()` evaluates to false for which we still want to set a value @andrewlee94  @bknueven ?

Reply:
The error message below speaks about mutable Params, which have a `value` attribute but no `fixed` attribute. To maintain the existing behavior when `overwrite_fixed` is `False` we should do something like
```suggestion
                if overwrite_fixed or not(v[i].is_variable_type() and v[i].fixed):
```

Reply:
Very good. Hopefully @eslickj will add a test to ensure this behavior is maintained.

Reply:
From an IDAES specific stand-point, I have to ask why a `Param` would appear in a `Port` - this would tend to indicate a (very big) modeling error on the part of the user. That said, that is something that should be caught well before this point, which would mean when the `Port` is constructed.

Reply:
I guess my feeling (and I was too lazy to test this) is that named expressions have a value and it can be set.  It is deprecated and produces a warning message, but it still works.  I'll a little worried that if we don't explicitly limit this to variables or variables and parameters if that makes sense, someone could run into random weirdness at some point.

Reply:
I should say I do use expressions in ports and probably the reason I'm not aware of this causing trouble is that the arrangement I have is probably always from expression to variable.

Reply:
@eslickj If you are using `Expressions` in `Ports`, that is the one time to use the `VarLikeExpression` component from IDAES as it was built specifically to address these issues. E.g. it overloads the `set_value` method to raise an `Exception` to prevent this from occurring. Also, `Expression` (or `VarLikeExpression` should (probably?) only be used in outlets, and not inlets because of this.

Reply:
@andrewlee94 I do always use them in outlets, and I'm sure that's why I don't seem to have trouble with this.  Most properties aside from enthalpy, pressure, flow and composition in the Helmholtz EoS module are expressions.  I don't know what a person may put in a port or the direction they may think of their arcs going.  I guess a person could do something reasonable that we may think is weird.  That may argue for using VarLikeExpression for all expressions at least.  Maybe guarding against overwriting expressions here.

Reply:
If we abandon the duck-typing with this change (as it is) the exception below will only be raised if the port has some non-Pyomo like object on it or the object incorrectly reports itself as a variable type. If the object implements `is_variable_type` and it returns `False`, it'll simply be ignored.

So if we expect the destinations to be variables I'd move the check `is_variable_type` up raise an exception if it's `False`, to avoid a perhaps unexpected silent failure.